pkg/metrics/metrics.jsx

```diff
@@ -486,64 +583,75 @@ class CurrentMetrics extends React.Component {
-                </Button>
-            );
-            if (is_container && !podman_installed) {
-                name_text = (
-                    <Tooltip content={_("cockpit-podman is not installed")} key={name + "_tooltip"}>
-                        <div>
-                            {name_text}
-                        </div>
-                    </Tooltip>);
-            }
-            const value_text = <TableText wrapModifier="nowrap">{value}</TableText>;
-            return {
-                cells: [{ title: name_text }, { title: value_text }]
-            };
-        }
-
         // top 5 CPU and memory consuming systemd units
         const topServicesCPU = n_biggest(this.cgroupCPUNames, this.samples[9], 5);
         newState.topServicesCPU = topServicesCPU.map(
-            ([key, value, is_user, is_container, userid]) => cgroupRow(key, Number(value / 10 / numCpu).toFixed(1), is_user, is_container, userid) // usec/s → percent
+            ([key, value, is_user, is_container, userid]) => this.cgroupRow(key, Number(value / 10 / numCpu).toFixed(1), is_user, is_container, userid) // usec/s → percent
         );
 
         const topServicesMemory = n_biggest(this.cgroupMemoryNames, this.samples[10], 5);
         newState.topServicesMemory = topServicesMemory.map(
-            ([key, value, is_user, is_container, userid]) => cgroupRow(key, cockpit.format_bytes(value), is_user, is_container, userid)
+            ([key, value, is_user, is_container, userid]) => this.cgroupRow(key, cockpit.format_bytes(value), is_user, is_container)
         );
 
-        const notMappedContainers = topServicesMemory.concat(topServicesCPU).filter(([key, value, is_user, is_container, userid]) => is_container && getCachedPodName(userid, key) === undefined);
+        const notMappedContainers = topServicesMemory.concat(topServicesCPU).filter(([key, value, is_user, is_container, userid]) => is_container && this.getCachedPodName(userid, key) === undefined);
         if (notMappedContainers.length !== 0) {
             this.update_podman_name_mapping(notMappedContainers);
         }
         this.setState(newState);
     }
 
+    getCachedPodName = (uid, containerid) => this.state.podNameMapping[uid] && this.state.podNameMapping[uid][containerid];
+
+    cgroupRow = (name, value, is_user, is_container, uid, valueB) => {
+        // valueB: optional parameter, used for cgroup disk write
+        const podman_installed = cockpit.manifests && cockpit.manifests.podman;
+
+        const cgroupClickHandler = (name, isUser, isContainer, uid) => {
+            if (isContainer) {
+                const containerName = this.getCachedPodName(uid, name);
+                if (containerName) {
+                    cockpit.jump("/podman#/?name=" + containerName);
+                } else {
+                    cockpit.jump("/podman");
+                }
+            } else {
+                cockpit.jump("/system/services#/" + name + ".service" + (isUser ? "?owner=user" : ""));
```

Comment:
These 9 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-17692-20230130-084654-b6542dcc-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)